src/stack-overview.ts

```diff
@@ -0,0 +1,254 @@
+
+  const lastModified = new Date(lastModifiedStr);
+
+  console.debug(
+    `stack:${
+      metadata.StackName ?? "unknown"
+    }; lambda:${id}; lastModified:${lastModified.toISOString()}; oneMonthAgo:${sixMonthsAgo.toISOString()}`
+  );
+
+  return lastModified < sixMonthsAgo ? "NoRecentLambdaDeployment" : undefined;
+};
+
+const noRecentAsgDeployment = async (
+  clients: Clients,
+  metadata: StackMetadata
+): Promise<InactiveStatus | undefined> => {
+  const asg = Object.entries(metadata.Template.Resources).find(
+    ([, resource]) => resource.Type === "AWS::AutoScaling::AutoScalingGroup"
+  );
+
+  if (!asg) return undefined;
+
+  const [id] = asg;
+
+  const describeStackCmd = new DescribeStackResourceCommand({ StackName: metadata.StackName, LogicalResourceId: id });
+  const describeStackOutput = await clients.cfnClient.send(describeStackCmd);
+  const asgId = describeStackOutput.StackResourceDetail?.PhysicalResourceId;
+  const describeScalingActivitiesCmd = new DescribeScalingActivitiesCommand({ AutoScalingGroupName: asgId });
+  const describeScalingActivitiesOutput = await clients.asgClient.send(describeScalingActivitiesCmd);
+  const activities = describeScalingActivitiesOutput.Activities ?? [];
+
+  const userRequests = activities.filter((activity) => activity.Cause?.includes("user request"));
+  if (userRequests.length === 0) return "NoRecentAsgDeployment";
+
+  const lastUserRequest = userRequests[0].StartTime;
+  if (!lastUserRequest) return "NoRecentAsgDeployment";
+
+  const oneMonthAgo = new Date();
+  oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
+
+  console.debug(
+    `stack:${
+      metadata.StackName ?? "unknown"
+    }; asg:${id}; lastActivity:${lastUserRequest.toISOString()}; oneMonthAgo:${oneMonthAgo.toISOString()}`
+  );
+
+  return lastUserRequest < oneMonthAgo ? "NoRecentAsgDeployment" : undefined;
+};
+
+type Clients = {
+  asgClient: AutoScalingClient;
+  cfnClient: CloudFormationClient;
+  lambdaClient: LambdaClient;
+};
+
+export const run = async (profile: string, region: string, preferCache: boolean) => {
+  // filter stacks to ASG and lambda
+  const isStandardService = (metadata: StackMetadata): boolean => {
+    return !!Object.entries(metadata.Template.Resources).find(
+      ([, resource]) =>
+        resource.Type === "AWS::AutoScaling::AutoScalingGroup" || resource.Type === "AWS::Lambda::Function"
+    );
+  };
+
+  const stacks = await getStacks([profile], [region], preferCache);
+  const clients = {
+    cfnClient: new CloudFormationClient({
+      region,
+      maxAttempts: Config.SDK_MAX_ATTEMPTS,
+      credentials: fromIni({ profile }),
+    }),
+    asgClient: new AutoScalingClient({
+      region,
+      maxAttempts: Config.SDK_MAX_ATTEMPTS,
+      credentials: fromIni({ profile }),
+    }),
+    lambdaClient: new LambdaClient({
+      region,
+      maxAttempts: Config.SDK_MAX_ATTEMPTS,
+      credentials: fromIni({ profile }),
+    }),
+  };
```

Comment:
Does this require more permissions than the CFN Read ones from before? I wonder if this repo is moving in the self-serve direction, rather than DevX produces a spreadsheet for others to consume?

Reply:
Hmm, it's a good point. I'm not sure how best to consume this to be honest. We could link it to the CDK getting started docs somehow?

Reply:
Happy to defer this.